app/depict/src/main/java/org/openscience/cdk/depict/Dimensions.java

```diff
@@ -0,0 +1,155 @@
+/* Copyright (C) 2015  The Chemistry Development Kit (CDK) project
+ *
+ * Contact: cdk-devel@lists.sourceforge.net
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License
+ * as published by the Free Software Foundation; either version 2.1
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+package org.openscience.cdk.depict;
+
+import org.openscience.cdk.renderer.elements.Bounds;
+
+import java.awt.Dimension;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.TreeSet;
+
+/**
+ * Internal: Immutable value class to help with store diagram dimensions
+ * (a tuple width and height). Given some dimensions we can add/subtract,
+ * grow/shrink as needed. Utility methods are provided for laying out rendering
+ * elements in grids and rows.
+ */
+final class Dimensions {
+
+    /**
+     * Magic value for automated sizing.
+     */
+    final static Dimensions AUTOMATIC = new Dimensions(DepictionGenerator.AUTOMATIC,
+                                                       DepictionGenerator.AUTOMATIC);
+
+    /**
+     * The values.
+     */
+    final double w, h;
+
+    public Dimensions(double w, double h) {
```

Comment:
We need to think how to get PMD accept a selection of single char variable names... alternatively, use @NOPMD or so... see the PMD website...